lib/publiccloud/azure_client.pm

```diff
@@ -62,6 +64,45 @@ sub vault_create_credentials {
     }
 }
 
+=head2 configure_docker
+
+Configure the docker to access the cloud provider registry
+=cut
+
+sub configure_docker {
+    my ($self) = @_;
+
+    my $login_cmd = sprintf(q(while ! az acr login --name '%s' -u '%s' -p '%s'; do sleep 10; done),
+        $self->container_registry, $self->key_id, $self->key_secret);
+    assert_script_run($login_cmd);
+    $login_cmd = sprintf(q(docker login %s.azurecr.io), $self->container_registry);
+    assert_script_run($login_cmd);
+}
+
+=head2 get_container_registry_prefix
+
+Get the full registry prefix URL (based on the account and region) to push container images on ACR.
+=cut
+
+sub get_container_registry_prefix {
+    my ($self) = @_;
+    my $full_name_prefix = sprintf('%s.azurecr.io', $self->container_registry);
+    return $full_name_prefix;
+}
+
+=head2 get_container_image_full_name
+
+Returns the full name of the container image in ACR registry
+C<tag> Tag of the container
+=cut
+
+sub get_container_image_full_name {
+    my ($self, $tag) = @_;
+    my $full_name_prefix = $self->get_container_registry_prefix();
+    return "$full_name_prefix/$tag";
+}
```

Comment:
i think `get_container_registry_prefix` and `get_container_image_full_name` could be one single function. wdyt?

Reply:
I think we discuss that in the past with someone else. The is spitted in two because get_container_registry_prefix is used directly

Reply:
i didnt found a ref in this PR before my comment but i will trust you for now.

Reply:
yes, I checked it and you are right! is not needed. but if you don't mind I created a ticket for that https://progress.opensuse.org/issues/104229